scripts/index_test_examples.py

```
@@ -0,0 +1,91 @@
+
+def extract_class_name(java_code):
+    match = re.search(r'class\s+(\w+)', java_code)
+    return match.group(1) if match else None
+
+def extract_method_signatures(java_code):
+    pattern = re.compile(r'(public|protected|private)\s+[\w<>,\[\]]+\s+(\w+)\s*\([^)]*\)')
+    return [m.group(0) for m in pattern.finditer(java_code)]
+
+# --- MAIN ---
+def main():
+    print(f"[RAG INDEX] Scanning for test/source pairs...")
```

Ruff (0.11.9)

f-string without any placeholders

Remove extraneous `f` prefix

(F541)